### PR TITLE
MAKE-533: create pail evergreen project/config/make system

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: evergreen.yaml [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+      - func: setup-credentials
+      - func: run-make
+        vars: { target: "${task_name}" }
+
+
+#######################################
+#              Functions              #
+#######################################
+functions:
+  setup-credentials:
```
[the remark this answers is not in the document]
Comment (julianedwards) [reply evergreen.yaml]: You want to use subprocesses.exec instead?
https://github.com/evergreen-ci/evergreen/blob/master/self-tests.yml#L145

File: makefile [hunk 2]
```diff
@@ -2,7 +2,63 @@ buildDir := build
...
+#   difficult to test without a much more complex reflection/code
+#   generation approach, so we ignore dupl errors in tests.
+lintArgs += --exclude="warning: duplicate of .*_test.go"
+#   go lint warns on an error in docstring format, erroneously because
+#   it doesn't consider the entire package.
+lintArgs += --exclude="warning: package comment should be of the form \"Package .* ...\""
+#   known issues that the linter picks up that are not relevant in our cases
+lintArgs += --exclude="file is not goimported" # top-level mains aren't imported
+lintArgs += --exclude="error return value not checked .defer.*"
+lintArgs += --exclude="\w+Key is unused.*"
```
Comment (tychoish) [inline makefile]: delete this line and the next two.

File: makefile [hunk 3]
```diff
@@ -2,7 +2,63 @@ buildDir := build
...
+#   package, testing, and linter dependencies specified
+#   separately. This is a temporary solution: eventually we should
+#   vendorize all of these dependencies.
+lintDeps := github.com/alecthomas/gometalinter
+#   include test files and give linters 40s to run to avoid timeouts
+lintArgs := --tests --deadline=13m --vendor
+#   gotype produces false positives because it reads .a files which
+#   are rarely up to date.
+lintArgs += --disable="gotype" --disable="gosec" --disable="gocyclo" --disable="golint"
+lintArgs += --disable="megacheck" --enable="unused" --enable="gosimple"
```
Comment (tychoish) [inline makefile]: we can probably enable golint and megacheck. it's ok if the lint fails at first.

File: makefile [hunk 4]
```diff
@@ -2,7 +2,63 @@ buildDir := build
...
+lintArgs += --skip="build"
+#   enable and configure additional linters
+lintArgs += --line-length=100 --dupl-threshold=150 --cyclo-over=15
```
Comment (tychoish) [inline makefile]: I'd leave cyclo at the defauilt

File: makefile [hunk 5]
```diff
@@ -2,7 +2,63 @@ buildDir := build
...
+#   the gotype linter has an imperfect compilation simulator and
+#   produces the following false postive errors:
+lintArgs += --exclude="error: could not import github.com/mongodb/greenbay"
```
Comment (tychoish) [inline makefile]: delete line 29-31

**Pull request merged**